src/parse.rs

```diff
@@ -105,21 +88,49 @@ mod tests {
     use super::*;
 
     #[test]
-    fn test_get_file_extension_from_diff_line() {
+    fn test_get_file_extension_from_marker_line() {
         assert_eq!(
-            get_file_extension_from_diff_line("diff --git a/src/main.rs b/src/main.rs"),
+            get_file_extension_from_marker_line(
+                "--- src/one.rs	2019-11-20 06:47:56.000000000 +0100"
+            ),
             Some("rs")
         );
     }
 
     #[test]
-    fn test_get_file_extension_from_marker_line() {
+    fn test_get_file_extension_from_file_meta_line() {
         assert_eq!(
-            get_file_extension_from_marker_line(
-                "--- src/one.rs	2019-11-20 06:47:56.000000000 +0100"
-            ),
+            get_file_extension_from_file_meta_line_file_path("a/src/parse.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("b/src/pa rse.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("src/pa rse.rs"),
+            Some("rs")
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("wat hello.rs"),
             Some("rs")
         );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("/dev/null"),
+            None
+        );
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("Makefile"),
+            Some("Makefile")
+        );
```

Comment:
Before I write anything else I want to make it clear that you've already done more than enough for your PR to be merged and it's very helpful, so thank you!

I don't have time right now to do further work on these edge cases myself so I'm just going to make so additional comments on the off chance that you want to double-check my thinking, since it's very helpful to have someone else familiar with this code. 

Would you agree that this test case demonstrates a bug? I wonder if what we should do (in a different PR) is determine (programatically?) the cases where the sublime syntax definition is keyed off the entire filename (like `Makefile`, `Dockerfile`) and otherwise refuse to identify a file extension for random filenames that do not look like `*.foo`. 

```diff
diff --git a/src/parse.rs b/src/parse.rs
index e1a0860..78cea85 100644
--- a/src/parse.rs
+++ b/src/parse.rs
@@ -123,6 +123,11 @@ mod tests {
             get_file_extension_from_file_meta_line_file_path("Makefile"),
             Some("Makefile")
         );
+        // TODO: The fact that the following test passes is perhaps a bug.
+        assert_eq!(
+            get_file_extension_from_file_meta_line_file_path("rs"),
+            Some("rs")
+        );
         assert_eq!(
             get_file_extension_from_file_meta_line_file_path("a/src/Makefile"),
             Some("Makefile")
```

Reply:
Oh, I see. Nice catch.
I'm not sure if there is a good way to differentiate them unless we have a list of all file types that don't have an extension e.g. `Makefile`, `Dockerfile`.

Reply:
I was hoping that the sublime syntax definitions might distinguish between genuine extensions, that involve a `.`, like `foo.py`, vs whole-file-name extensions like `Makefile`, `Dockerfile`. However, it looks like they don't. So I don't think this is delta's problem, seeing as bat also suffers from it:

 <table><tr><td><img width=300px src="https://user-images.githubusercontent.com/52205/80925808-cd5ca880-8d60-11ea-988f-cdd7a88d9f23.png" alt="image" /></td></tr></table>